src/arrayMethodIndexOf.js

```diff
@@ -6,8 +6,34 @@
  * also work for negative fromIndex handling -2 as a pre last element.
  */
 function applyCustomIndexOf() {
-  [].__proto__.indexOf2 = function(searchElement, fromIndex) {
-    // write code here
+  [].__proto__.indexOf2 = function(searchElement, fromIndex = 0) {
+    let start = fromIndex;
+
+    if (this.length === 0
+      || fromIndex >= this.length
+      || arguments.length < 1) {
+      return -1;
+    }
+
+    if (start < 0) {
+      start = this.length + start + 1;
+    }
+
+    if (start < 0) {
+      start = 0;
+    }
+
+    for (let i = start; i <= this.length; i += 1) {
+      if (this[i] === searchElement
+        || (typeof searchElement === 'undefined'
+        && typeof this[i] === 'undefined')
+        || (Number.isNaN(searchElement)
+        && Number.isNaN(this[i]))) {
```

Comment:
the same

Reply:
can't leave && (...)  on the same line
then it's length > 80